app/controllers/application_controller.rb

```diff
@@ -1,5 +1,36 @@
 class ApplicationController < ActionController::Base
-  # Prevent CSRF attacks by raising an exception.
-  # For APIs, you may want to use :null_session instead.
   protect_from_forgery with: :exception
+
+  def set_flash(msg)
+    flash[:notice] = msg
+  end
+
+  def current_user
+    session[:user_id] ? User.find(session[:user_id]) : nil
```

Comment:
Find_by probably better here to avoid raising  a not found error where session[:user_id] exists but is not a valid user.